src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -153,22 +171,40 @@ Datum
 create_reference_table(PG_FUNCTION_ARGS)
 {
 	Oid relationId = PG_GETARG_OID(0);
+
+	CreateReferenceTable(relationId);
+
+	PG_RETURN_VOID();
+}
+
+
+/*
+ * TODO: add comment
+ */
+static void
+CreateReferenceTable(Oid relationId)
+{
+	uint32 colocationId = INVALID_COLOCATION_ID;
+	List *workerNodeList = WorkerNodeList();
 	int shardCount = 1;
-	AttrNumber firstColumnAttrNumber = 1;
+	int replicationFactor = list_length(workerNodeList);
+	Oid distributionColumnType = InvalidOid;
+	char *distributionColumnName = NULL;
 
-	char *firstColumnName = get_attname(relationId, firstColumnAttrNumber);
-	if (firstColumnName == NULL)
+	/* check for existing colocations */
+	colocationId = ColocationId(shardCount, replicationFactor, distributionColumnType);
+	if (colocationId == INVALID_COLOCATION_ID)
 	{
-		char *relationName = get_rel_name(relationId);
-		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-						errmsg("reference table candidate %s needs to have at"
-							   "least one column", relationName)));
+		colocationId = CreateColocationGroup(shardCount, replicationFactor,
+											 distributionColumnType);
 	}
 
-	CreateHashDistributedTable(relationId, firstColumnName, shardCount,
-							   ShardReplicationFactor);
+	/* first, convert the relation into distributed relation */
+	ConvertToDistributedTable(relationId, distributionColumnName,
```

Comment:
Which is relevant, since we use the repmodel to know which tables we can sync in MX.